### PR TITLE
don't subscribe queue to topic if subscription already exists

### PR DESCRIPTION
**Pull request by buzz1710**
localstack issue results in multiple duplicate messages if a queue subscribes to a topic multiple times:
https://github.com/localstack/localstack/issues/933

this prevents multiple subscribes from happening...

### CODE REVIEW

**Review by dcraig-sage — DISMISSED** [review]

File: lib/eventq/eventq_aws/aws_subscription_manager.rb [hunk 1]
```diff
@@ -72,6 +76,12 @@ def queue_policy(queue_arn)
           ]
         }'
       end
+
+      # check if there is an existing subscription for the given queue/topic
+      def existing_subscription?(queue_arn, topic_arn)
+        subscriptions = @client.sns.list_subscriptions.subscriptions
+        !subscriptions.select { |subscription| subscription.topic_arn == topic_arn && subscription.endpoint == queue_arn }.empty?
```
Comment (atothewest) [inline lib/eventq/eventq_aws/aws_subscription_manager.rb]: (Very minor) I think this could be `subscriptions.any? { ... }`

**Review by chrisbarber86 — DISMISSED** [review]

**Comment by tarastriska** [comment]
Hi, could you please merge it?

**Review by adamgeorgeson — DISMISSED** [review]

**Comment by chrisbarber86** [comment]
Do you want to bump the version file? https://github.com/Sage/eventq/blob/master/EVENTQ_VERSION
Suggest 2.4.0

**Review by chrisbarber86 — APPROVED** [review]

**Comment by adamgeorgeson** [comment]
Published in https://github.com/Sage/eventq/releases/tag/v2.4.0

**Pull request merged**